mockwebserver/src/main/kotlin/mockwebserver3/RecordedRequest.kt

```diff
@@ -74,9 +74,9 @@ class RecordedRequest(
   val handshakeServerNames: List<String>
 
   init {
-    if (socket is SSLSocket) {
+    if (socket is OkioSslSocket) {
       try {
-        this.handshake = socket.session.handshake()
+        this.handshake = socket.session?.handshake()
```

Comment:
Should we keep this as non-nullable, blow up if called in wrong state? or `socket.handshake` directly?

For discussion, just this is a chance to iron out the API a bit. But can be done later.

Reply:
Hmm, yeah. Looking at places that read `handshake`, basically everyone expects it to be non-null. From a very quick spot check, it looks like `socket.session` will effectively always be non-null, and so will `socket.session.handshake()`. So I think I'd be in favor of tightening up the constraints here. But probably in a second PR.